### PR TITLE
Adds aria-label to links that open in new windows

### LINKED ISSUE #4450: add aria-label="link text - new window" attribute in <a> tag when "open in new window" is checked
## Issue Overview

When a user checks "open in a new window" in link settings, add automatically aria-label attribute in the generated link  tag :
 `<a href="..." target="_blank" aria-label="link text - new window">link text</a>`

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Description
Adds an aria-lablel to the generated link when it is set to open in new window. Fixes #4450 

## How has this been tested?
Tested in Chrome for Mac OS
Ran `npm run test`

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/editor/src/components/rich-text/format-toolbar/link-container.js [hunk 1]
```diff
@@ -33,17 +33,29 @@ function getLinkAttributesFromFormat( { attributes: { href = '', target } = {} }
...
 			href,
 		},
 	};
 
+	const text = record.text || '';
+
 	if ( opensInNewWindow ) {
 		format.attributes.target = '_blank';
 		format.attributes.rel = 'noreferrer noopener';
+		format.attributes[ 'aria-label' ] = `${ text } (opens in a new tab)`;
```
Comment (ocean90) [inline packages/editor/src/components/rich-text/format-toolbar/link-container.js]: This should be translatable.

**Review by ocean90 — DISMISSED** [review]

**Comment by ryanwelcher** [comment]
@ocean90 I've updated the PR to address your feedback. Thanks for reviewing it !

File: packages/editor/src/components/rich-text/format-toolbar/link-container.js [hunk 2]
```diff
@@ -33,17 +33,29 @@ function getLinkAttributesFromFormat( { attributes: { href = '', target } = {} }
...
 			href,
 		},
 	};
 
+	const text = record.text || '';
+
 	if ( opensInNewWindow ) {
 		format.attributes.target = '_blank';
 		format.attributes.rel = 'noreferrer noopener';
+		format.attributes[ 'aria-label' ] = `${ text } (${ __( 'opens in a new tab' ) })`;
```
Comment (mcsf) [inline packages/editor/src/components/rich-text/format-toolbar/link-container.js]: This is still problematic from an i18n standpoint. Should be something like:

```js
// translators: accessibility label for external links, where the argument is the link text
const label = sprintf( __( '%s (opens in a new tab)' ), text );

format.attributes[ 'aria-label' ] = label;
```

Also note that an empty `text` leads to a label that starts with whitespace. Could be handled by calling `String#trim`: `sprintf( … ).trim()`.

**Comment by ryanwelcher** [comment]
@mcsf I've updated the PR based on your input. Thanks for the direction!

I've also made some changes to the `createLinkFormat` to better reflect the use of the 3rd parameter that was introduced in this PR.

**Review by mcsf — APPROVED** [review]
I pushed afd0191 straight from GitHub. Assuming that tests fine, this looks good to go.

**Comment by iseulde** [comment]
I see no tests. 😕

**Comment by aduth** [comment]
> As raised at [#11815 (comment)](https://github.com/WordPress/gutenberg/issues/11815#issuecomment-440275818), it seems the `aria-label` is including the full text of the RichText field in which its contained, when it seems like it ought to only contain the text of the link itself? Is that correct?

Created issue at #12325

**Pull request merged**